app/models/sushi.rb

```diff
@@ -1,14 +1,23 @@
 # frozen_string_literal:true
 
 module Sushi
+  class Sushi::NotFoundError < StandardError
+    class << self
+      def no_records_within_date_range
+        new('There are no results for the given date range.')
+      end
+    end
+  end
+
+
   ##
   # Raised when the parameter we are given does not match our expectations; e.g. we can't convert
   # the text value to a Date.
   class InvalidParameterValue < StandardError
     # rubocop:disable Metrics/LineLength
     class << self
       def invalid_item_id(item_id)
-        new("The given parameter `item_id=#{item_id}` did not return any results. Either there are no metrics for this id during the dates specified, or there are no metrics for this id at all. Please confirm all given parameters.")
+        new("The given parameter `item_id=#{item_id}` is invalid. Please provide a valid item_id, or none at all.")
```

Comment:
This kind of exception should probably bubble up as a 404 error.  I'd rework this to also be a `Sushi::NotFoundError`.

Because they provided an item_id and that doesn't exist.  This maps to the default behavior of `ActiveRecord::NotFoundError`, in that when you request an ID that does not exist, it raises an `ActiveRecord::NotFoundError` but Rails by default rescues that to 404.